liblangutil/SemVerHandler.cpp

```diff
@@ -208,9 +208,18 @@ void SemVerMatchExpressionParser::parseMatchExpression()
 	range.components.push_back(parseMatchComponent());
 	if (currentToken() == Token::Sub)
 	{
-		range.components[0].prefix = Token::GreaterThanOrEqual;
 		nextToken();
 		range.components.push_back(parseMatchComponent());
+
+		if (TokenTraits::isPragmaOp(range.components[0].prefix) || TokenTraits::isPragmaOp(range.components[1].prefix))
+		{
+			solThrow(
+				SemVerError,
+				"You cannot use operators (<, <=, >=, >, ^) with version ranges (-)."
+			);
+		}
```

Comment:
I don't think it's a good idea to check only for specific prefixes. For example currently this condition misses `~`, which is also a prefix. Ranges can have no prefixes so we should just error out here on any of them. I'd also make the message just that:

```
Components of version ranges cannot be prefixed with operators.
```
Might be also a good idea to show which token it was. You can use `TokenTraits::friendlyName(token)` for that.

Reply:
How could I detect if it the range has a prefix? 

At this point after calling the function `parseMatchComponent()` the `range.components[0].prefix` and `range.components[1].prefix` will at least have the default prefix (Assign) making impossible to know if it was assigned because it really has the prefix `=` or is the default one.

I think I could change it but it would modify the function `parseMatchComponent()`, a bigger change at the code.

On the other hand I could just add the prefix `~` at the condition.

As you decide, I await a reply! :)

Reply:
Good point, but it's a separate problem. The current code has it too. It does not discern between `=` and having no prefix either. It's actually wrong, which I did not notice until you pointed it out. It will treat `=0.0 - =1.0` as a valid range (and by the way, you should have a test case covering it).

As for how to distinguish then, I think we should change `prefix` to `optional<Token>` and not set the default in `parseMatchComponent()`. Then handle `nullopt` explicitly in `matches()`.